controllers/services/auth/auth_controller.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"fmt"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	componentsApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1alpha1"
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/actions/render/template"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/handlers"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/predicates"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/reconciler"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+const (
+	odhDashboardConfigCRDName = "odhdashboardconfigs.opendatahub.io"
+)
+
+// NewServiceReconciler creates a ServiceReconciler for the Auth API.
+func NewServiceReconciler(ctx context.Context, mgr ctrl.Manager) error {
+	_, err := reconciler.ReconcilerFor(mgr, &serviceApi.Auth{}).
+		// operands - owned
+		Owns(&rbacv1.ClusterRoleBinding{}).
+		Owns(&rbacv1.ClusterRole{}).
+		Owns(&rbacv1.Role{}).
+		Owns(&rbacv1.RoleBinding{}).
+		WatchesGVK(gvk.Dashboard).
+		WatchesGVK(
+			gvk.CustomResourceDefinition,
+			reconciler.WithEventHandler(handlers.ToNamed(serviceApi.AuthInstanceName)),
+			reconciler.WithPredicates(predicate.NewPredicateFuncs(func(object client.Object) bool {
+				return object.GetName() == odhDashboardConfigCRDName
+			}))).
+		WatchesGVK(
+			gvk.OdhDashboardConfig,
+			reconciler.Dynamic(shouldWatchDashboardConfig),
```

Comment:
if we are locking down changes in odh-dashboard-config (as read-only) for groupsConfig block, do we still need to watch this object? as if the change is done for .spec.dashboardConfig.disalbeModelMesh should reconcile be triggered.

Reply:
We have two options:
1. we do sync `OdhDashbaordConfig` only once when creating the `Auth` instance
2. we do sync `OdhDashbaordConfig` as part of the reconcile loop. 

As today, it is done as part of the reconcile loop (2), so we must watch the resource to be consistent, otherwise the sync would happen by chance i.e, if the controller gets restarted. I guess that when the auth stanza in the `OdhDashbaordConfig` will be made read-only, we can probably switch to option (1), but that's something for @StevenTobin and @csams to clarify.

Reply:
yeah thats my understanding after yesterday's internal clairification.
was not aware of the "read-only" set on the groupsConfig previously

Reply:
Doing it once was problematic, because we can't guarentee it would there so we would have to check for creation anyway. Doing it as part of reconciliation means we can avoid edge cases where the dashboardConfig may be created later or edited and recreated later. 

Also remember this watch is only neccessary as part of the migration period and dashboard will be deprecating the groupsConfig from the dashboardConfig at which point this watch can be removed.